app/storage/encrypted_storage.py

```diff
@@ -5,80 +5,48 @@
 from app import settings
 from app.cryptography.jwe_decryption import JWEDirDecrypter
 from app.cryptography.jwe_encryption import JWEDirEncrypter
-from app.storage.abstract_server_storage import AbstractServerStorage
+from app.storage.database_storage import DatabaseStorage
 from app.utilities.strings import to_bytes
 from app.utilities.strings import to_str
 
 logger = logging.getLogger(__name__)
 
 
-class EncryptedServerStorageDecorator(AbstractServerStorage):
+def generate_key(user_id, user_ik, pepper=settings.EQ_SERVER_SIDE_STORAGE_ENCRYPTION_KEY_PEPPER):
+    sha256 = hashlib.sha256()
+    sha256.update(to_str(user_id).encode('utf-8'))
+    sha256.update(to_str(user_ik).encode('utf-8'))
+    sha256.update(to_str(pepper).encode('utf-8'))
 
-    JSON_DATA_KEY = 'data'
+    # we only need the first 32 characters for the CEK
+    cek = sha256.hexdigest()[:32]
+    return to_bytes(cek)
 
-    def __init__(self, server_storage):
+
+class EncryptedStorage(DatabaseStorage):
```

Comment:
Had a look at master and doesn't appear to trip up there, presumably because the abstract class wasn't using a proper base class and the signatures didn't appear as different. Will have to have a think on what the best approach here is.